client/jetpack-connect/authorize.js

```diff
@@ -660,10 +662,11 @@ export default connect(
 			isFetchingAuthorizationSite: isRequestingSite( state, authQuery.clientId ),
 			isFetchingSites: isRequestingSites( state ),
 			isMobileAppFlow,
+			isVip: isVipSite( state, authQuery.clientId ),
```

Comment:
We should have a look at some history and try to figure out what's going on.

Is the `vip` plan now the canonical way for detecting a VIP site?

Maybe this needs to be updated since #13400 (@tyxla) and the attribute could be removed as suggested in #14096.

@brettshumaker I'd suggest digging around more to determine why you aren't seeing this attribute set for a specific VIP site.

Reply:
@sirreal Ok. I've done some digging this morning and have found that `is_vip()` in [this file]( https://opengrok.a8c.com/source/xref/trunk/public.api/rest/sal/class.json-api-site-jetpack-base.php#105) is the reason `is_vip` is set to `false` on VIP Go sites in Calypso. I'm having difficulty tracing the history of this function, however. It looks like this file was moved to its current location in early 2016 and I'm not able to trace it back to where it was prior to that to see if there are any clues as to why its was just returning `false`.

Changing that function to return `wpcom_is_vip( $this->blog_id );` is giving me expected results in Calypso and allowing me to be redirected back to the URL provided during the Jetpack connection flow. 

> Is the vip plan now the canonical way for detecting a VIP site?

Probably not. `/public.api/rest/wpcom-json-endpoints/class.wpcom-json-api-me-sites-v1-2-endpoint.php` is using `wpcom_is_vip()` to alter the Jetpack plan information for VIP sites, so I suspect that should be the correct way.